sdk/include/platform/sunburst/platform-rgbctrl.hh

```diff
@@ -0,0 +1,83 @@
+#pragma once
+#include <cdefs.h>
+#include <stdint.h>
+
+/**
+ * An enum representing each of the Sonata's RGB LEDs.
+ */
+enum class SonataRgbLed
+{
+	Led0,
+	Led1,
+};
+
+/**
+ * A driver for the Sonata's RGB LED Controller
+ */
+struct SonataRgbLedController
+{
+	/**
+	 * Registers for setting the 8-bit red, green, and blue values
+	 * for the two RGB Leds.
+	 */
+	uint32_t rgbLed0;
+	uint32_t rgbLed1;
+	/// Control Register
+	uint32_t control;
+	/// Status Register
+	uint32_t status;
+
+	/// Control Register Fields
+	enum [[clang::flag_enum]] : uint32_t{
+	  /// Write 1 to set RGB LEDs to specified colours.
+	  ControlSet = 1 << 0,
+	  /**
+	   * Write 1 to turn off RGB LEDs.
+	   * Write to ControlSet to turn on again.
+	   */
+	  ControlOff = 1 << 1,
+	};
+
+	/// Status Register Fields
+	enum [[clang::flag_enum]] : uint32_t{
+	  /**
+	   * When asserted controller is idle and new colours can be set,
+	   * otherwise writes to regLed0, regLed1, and control are ignored.
+	   */
+	  StatusIdle = 1 << 0,
+	};
+
+	void wait_for_idle() volatile
```

Comment:
What does idle mean for an LED?  I vaguely remember you or Greg saying that these are surprisingly stateful devices and you're sending them update commands, but a comment to that effect would be nice.

Reply:
Does it ever become busy as a result of something external (and so is this racy)?  I'm assuming not and this is similar to the colour LED strips where you just write a load of 24-bit values in a waveform down the wire to update each one and so the controller here is just a little queue that stores the two words and writes them down the wire?

Reply:
Yes, I don't think it can be a result of something external.

(Idle is assigned [here](https://github.com/lowRISC/sonata-system/blob/191f5a796e4fcc032c6325d1eb1122150e23d0db/rtl/ip/rgbled_ctrl/rtl/ws281x_drv.sv#L134).)